pkg/jx/cmd/add_app.go

```diff
@@ -80,38 +87,57 @@ func (o *AddAppOptions) addFlags(cmd *cobra.Command, defaultNamespace string, de
-	cmd.Flags().StringVarP(&o.ReleaseName, optionRelease, "r", defaultOptionRelease, "The chart release name [--no-gitops]")
-	cmd.Flags().BoolVarP(&o.HelmUpdate, "helm-update", "", true, "Should we run helm update first to ensure we use the latest version [--no-gitops]")
-	cmd.Flags().StringVarP(&o.Namespace, "namespace", "n", defaultNamespace, "The Namespace to install into [--no-gitops]")
-	cmd.Flags().StringArrayVarP(&o.ValueFiles, "values", "f", []string{}, "List of locations for values files, can be local files or URLs [--no-gitops]")
-	cmd.Flags().StringVarP(&o.SetValues, "set", "s", "",
-		"The chart set values (can specify multiple or separate values with commas: key1=val1,key2=val2) [--no-gitops]")
+	cmd.Flags().StringVarP(&o.Alias, optionAlias, "", "",
+		"An alias to use for the app (available when using GitOps for your dev environment)")
+	cmd.Flags().StringVarP(&o.ReleaseName, optionRelease, "r", defaultOptionRelease,
+		"The chart release name (available when NOT using GitOps for your dev environment)")
+	cmd.Flags().BoolVarP(&o.HelmUpdate, optionHelmUpdate, "", true,
+		"Should we run helm update first to ensure we use the latest version (available when NOT using GitOps for your dev environment)")
+	cmd.Flags().StringVarP(&o.Namespace, optionNamespace, "n", defaultNamespace, "The Namespace to install into (available when NOT using GitOps for your dev environment)")
+	cmd.Flags().StringArrayVarP(&o.ValueFiles, optionValues, "f", []string{}, "List of locations for values files, "+
+		"can be local files or URLs (available when NOT using GitOps for your dev environment)")
+	cmd.Flags().StringVarP(&o.SetValues, optionSet, "s", "",
+		"The chart set values (can specify multiple or separate values with commas: key1=val1,key2=val2) (available when NOT using GitOps for your dev environment)")
 
 }
 
 // Run implements this command
 func (o *AddAppOptions) Run() error {
 	o.GitOps, o.DevEnv = o.GetDevEnv()
 	if o.Repo == "" {
-		o.Repo = o.DevEnv.Spec.TeamSettings.AppsRepository		
+		o.Repo = o.DevEnv.Spec.TeamSettings.AppsRepository
 	}
 
-	// Regiser the App CRD
-	apiClient, err := o.CreateApiExtensionsClient()
-	if err != nil {
-		return err
+	if o.GitOps {
+		msg := "Unable to specify --%s when using GitOps for your dev environment"
+		if o.Cmd.Flag(optionRelease).Changed {
+			return util.InvalidOptionf(optionRelease, o.ReleaseName, msg, optionRelease)
+		}
+		if o.Cmd.Flag(optionHelmUpdate).Changed {
+			return util.InvalidOptionf(optionHelmUpdate, o.ReleaseName, msg, optionHelmUpdate)
+		}
+		if o.Cmd.Flag(optionNamespace).Changed {
+			return util.InvalidOptionf(optionNamespace, o.ReleaseName, msg, optionNamespace)
+		}
+		if o.Cmd.Flag(optionValues).Changed {
+			return util.InvalidOptionf(optionValues, o.ReleaseName, msg, optionValues)
+		}
+		if o.Cmd.Flag(optionSet).Changed {
+			return util.InvalidOptionf(optionSet, o.ReleaseName, msg, optionSet)
+		}
 	}
-	err = kube.RegisterAppCRD(apiClient)
```

Comment:
I guess these calls were obsolete?

Reply:
Yes, we are trying to bulk register CRDs upfront now.